.github/workflows/main.yml

```
@@ -0,0 +1,17 @@
+name: Test backend
+on: [push]
```

Credits to @one-jonathan , this way we only run the test when pushing directly to main (we agreed not to do this, but just in case), or when opening a PR to main. This seems sufficient, running the workflow every push seems a bit overkill. 
```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

---

In my personal opinion it's perfectly fine to run them on every push as long as it doesn't take 10 minutes or uses expensive resources. The earlier you find introduced bugs the easier they are to fix